File: frontend/src/types/internal-chat.ts
import type { CargoUsuario } from "./index";

export type TipoCanalChatInterno = "GERAL" | "DIRETO" | "GRUPO";

export interface ChatInternoAutor {
  id: number;
  nome: string;
  cargo: CargoUsuario;
  foto_perfil: string | null;
  ativo: boolean;
}

export type ChatInternoUsuario = ChatInternoAutor;

export interface ChatInternoMensagem {
  id: number;
  canal_id: number;
  conteudo: string;
  created_at: string;
  autor: ChatInternoAutor;
}

export interface ChatInternoCanal {
  id: number;
  tipo: TipoCanalChatInterno;
  nome: string;
  created_at: string;
  membros: ChatInternoAutor[];
  ultima_mensagem: ChatInternoMensagem | null;
  nao_lidas: number;
}

export interface ChatInternoResumo {
  nao_lidas: number;
  ultima_mensagem_id: number | null;
}

export interface ChatInternoLeitura {
  canal_id: number;
  ultima_mensagem_id: number;
  updated_at: string;
}

File: app/schemas/internal_chat.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import CargoUsuario


class ChatInternoMensagemCreate(BaseModel):
    conteudo: str = Field(min_length=1, max_length=2000)


class ChatInternoAutorOut(BaseModel):
    id: int
    nome: str
    cargo: CargoUsuario
    foto_perfil: str | None


class ChatInternoMensagemOut(BaseModel):
    id: int
    conteudo: str
    created_at: datetime
    autor: ChatInternoAutorOut


class ChatInternoResumoOut(BaseModel):
    nao_lidas: int
    ultima_mensagem_id: int | None


class ChatInternoLeituraOut(BaseModel):
    ultima_mensagem_id: int
    updated_at: datetime
